Yousif-Shaheen-submission/caesar.cpp

```diff
@@ -0,0 +1,83 @@
+#include <iostream>
+#include <string>
+#include <cctype>  
+
+
+const char alphabet_upper[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+const char alphabet_lower[] = "abcdefghijklmnopqrstuvwxyz";
+
+
+std::string encrypt(const std::string& word, int shift = 3) {
+    std::string final = "";
+
+    for (char letter : word) {
+        if (letter == ' ') {
+            final += ' ';
+            continue;
+        }
+
+        bool is_lower = std::islower(letter);
+
+       
+        char base_letter = is_lower ? std::tolower(letter) : letter;
+
+      
+        int index = base_letter - (is_lower ? 'a' : 'A');
+
+       
+        int encrypted_index = (index + shift) % 26;
+
+      
+        char encrypted_letter = (is_lower ? alphabet_lower : alphabet_upper)[encrypted_index];
+
+        final += encrypted_letter;
+    }
+
+    return final;
+}
+
+
+std::string decrypt(const std::string& word, int shift = 3) {
+    std::string final = "";
+
+    for (char letter : word) {
+        if (letter == ' ') {
+            final += ' ';
+            continue;
+        }
+
+        bool is_lower = std::islower(letter);
+
+       
+        char base_letter = is_lower ? std::tolower(letter) : letter;
+
+       
+        int index = base_letter - (is_lower ? 'a' : 'A');
+
+        
+        int decrypted_index = (index - shift + 26) % 26;
+
+       
+        char decrypted_letter = (is_lower ? alphabet_lower : alphabet_upper)[decrypted_index];
+
+        final += decrypted_letter;
+    }
+
+    return final;
+}
```

Comment:
You could also have written this as:
```suggestion
std::string decrypt(const std::string& word, int shift = 3) {
    return encrypt(word, -shift);
}
```